File: php/features.php
<?php
declare(strict_types=1);

// Newton SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NewtonFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NewtonBaseFeature();
            case "test":
                return new NewtonTestFeature();
            default:
                return new NewtonBaseFeature();
        }
    }
}

File: py/features.py
# Newton SDK feature factory

from feature.base_feature import NewtonBaseFeature
from feature.test_feature import NewtonTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NewtonBaseFeature(),
        "test": lambda: NewtonTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
